packages/react-scripts/config/webpack.config.prod.js

```diff
@@ -53,6 +56,18 @@ const extractTextPluginOptions = shouldUseRelativeAssetPaths
   ? // Making sure that the publicPath goes back to to build folder.
     { publicPath: Array(cssFilename.split('/').length).join('../') }
   : {};
+// Check if vendor file exists
+const checkIfVendorFileExists = fs.existsSync(paths.appVendorJs);
+// If the vendor file exists, add an entry point for vendor,
+// and a seperate entry for polyfills and app index file,
+// otherwise keep only polyfills and app index.
+const appEntryFiles = [require.resolve('./polyfills'), paths.appIndexJs];
```

Comment:
This way vendor bundle does not get the polyfills. Polyfills should the first thing that run from our bundle, else we could get issue similar to this https://github.com/facebook/react/issues/8379#issuecomment-263962787 for promises and fetch. I think we need another check here / move this to a resolver function.